packages/components/src/components/checkbox-group/checkbox-group.ts

```diff
@@ -0,0 +1,172 @@
+import '../icon/icon';
+import { css, html } from 'lit';
+import { customElement } from '../../../src/internal/register-custom-element';
+import { HasSlotController } from '../../internal/slot';
+import { property } from 'lit/decorators.js';
+import { watch } from '../../internal/watch';
+import componentStyles from '../../styles/component.styles';
+import cx from 'classix';
+import SolidElement from '../../internal/solid-element';
+import type SdCheckbox from '../checkbox/checkbox';
+
+/**
+ * @summary Checkbox groups are used to group multiple [checkbox](/components/checkbox). It provides only presentational functionality.
+ * @documentation https://solid.union-investment.com/[storybook-link]/checkbox-group
+ * @status stable
+ * @since 1.18.0
+ *
+ * @slot - The default slot where `<sd-checkbox>` elements are placed.
+ * @slot label - The checkbox group's label. Required for proper accessibility. Alternatively, you can use the `label`
+ * attribute.
+ **/
+
+@customElement('sd-checkbox-group')
+export default class SdCheckboxGroup extends SolidElement {
+  private readonly hasSlotController = new HasSlotController(this, 'label');
+
+  /**
+   * The checkbox group's label. Required for proper accessibility. If you need to display HTML, use the `label` slot
+   * instead.
+   */
+  @property() label = '';
+
+  /** The checkbox group's size. This size will be applied to the label, all child checkboxes. */
+  @property({ reflect: true }) size: 'lg' | 'sm' = 'lg';
+
+  /**
+   * The orientation property determines the alignment of the component's content or elements. It accepts two possible
+   * values: 'horizontal' and 'vertical'. The default value is 'vertical'.
+   * This property allows you to control the visual layout and arrangement of elements within the component, providing
+   * flexibility in how the component is displayed based on your specific design needs.
+   */
+  @property({ reflect: true }) orientation: 'horizontal' | 'vertical' = 'vertical';
+
+  private getAllCheckboxes() {
+    return [...this.querySelectorAll<SdCheckbox>('sd-checkbox')];
+  }
+
+  private async syncCheckboxElements() {
+    const checkboxes = this.getAllCheckboxes();
+
+    await Promise.all(
+      // Sync the checked state and size
+      checkboxes.map(async checkbox => {
+        await checkbox.updateComplete;
+
+        checkbox.size = this.size;
```

Comment:
I din't fully understand how you handle the invalid state from sd-checkbox to sd-checkbox-group.

Why haven't you created an 'invalid' property in 'sd-checkbox' and synced it like checkbox.size?

Reply:
Because the checkbox is a FormControl and has it's own validation. This is the main difference to the radio component. The radio component needs the radio-group component to pass in the invalid state.
We cannot do this for the checkbox-group and checkbox as we don't know what's valid and invalid in a checkbox-group.

Reply:
What @DanielHargesheimer says. The checkbox can and has to check itself. The check group can't know anything about validation, as it could be everything (check all cars in a list, check a max of 2, check a min of 2 etc.) and can only be handled by the application defining a validation.